src/Tribe/Integrations/WPML/Linked_Posts.php

```diff
@@ -137,23 +170,171 @@ protected function get_curent_language_linked_posts_ids( array $args ) {
+
+		return $linked_post_ids;
 	}
 
 	/**
+	 * Returns a list of linked post IDs for the default language.
+	 *
+	 *
 	 * @param array $args An array WP_Query arguments
 	 *
 	 * @return array An array of linked posts filtered by the default language
 	 */
 	protected function get_default_language_linked_post_ids( array $args ) {
+		$cache_key = $this->cache->make_key( func_get_args(), 'default_language_linked_post_ids' );
+		if ( isset( $this->cache[ $cache_key ] ) ) {
+			return $this->cache[ $cache_key ];
+		}
+
 		/** @var SitePress $sitepress */
 		global $sitepress;
+
 		$sitepress->switch_lang( $sitepress->get_default_language() );
 
 		$query = new WP_Query( $args );
 
 		$posts = $query->have_posts() ? $query->posts : array();
 
-		return array_filter( $posts, array( $this, 'is_not_translated' ) );
+		$sitepress->switch_lang( $sitepress->get_current_language() );
+
+		$not_translated = array_filter( $posts, array( $this, 'is_not_translated' ) );
+		$assigned = $this->get_linked_post_assigned_to_current( $args );
+
+		// if a linked post is assigned always show it, translated or not
+		$linked_post_ids = array_merge( $not_translated, $assigned );
+
+		$this->cache[ $cache_key ] = $linked_post_ids;
+
+		return $linked_post_ids;
+	}
+
+	/**
+	 * Returns the post ID(s) of post(s) of the type specified in the args linked to the current event.
+	 *
+	 * @param array $args An array of arguments in the format supported by `WP_Query`
+	 *
+	 * @return array An array of linked post IDs or an empty array if no post types, more than one post type
+	 *               is specified in the args, or the current post is not an event.
+	 */
+	protected function get_linked_post_assigned_to_current( array $args ) {
+		$post_type = (array) Tribe__Utils__Array::get( $args, 'post_type', array() );
+		$current_post_id = Tribe__Main::post_id_helper();
+
+		if ( ! tribe_is_event( $current_post_id ) ) {
+			return array();
+		}
+
+		if ( count( $post_type ) !== 1 || empty( $current_post_id ) ) {
+			return array();
+		}
+
+		$post_type = reset( $post_type );
+
+		$map = array(
+			Tribe__Events__Main::VENUE_POST_TYPE     => '_EventVenueID',
+			Tribe__Events__Main::ORGANIZER_POST_TYPE => '_EventOrganizerID',
+		);
+
+		if ( empty( $map[ $post_type ] ) ) {
+			return array();
+		}
+
+		$assigned = get_post_meta( $current_post_id, $map[ $post_type ], false );
+
+		return ! empty( $assigned ) ? $assigned : array();
+	}
+
+	/**
+	 * Conditionally sets up a `shutdown` action to translated the linked post IDs.
+	 *
+	 * @param array $data An array of data about the translation provided by WPML.
+	 *
+	 * @return bool Whether the `shutdown` action has been hooked or not.
+	 */
+	public function maybe_translate_linked_posts( array $data ) {
+		$required_keys = array( 'element_id', 'element_type', 'type' );
+
+		if ( count( array_intersect_key( $data, array_combine( $required_keys, $required_keys ) ) ) < count( $required_keys ) ) {
```

Comment:
Can you make sure this works on PHP 5.2? It normally doesn't like `count` with methods attached to it.